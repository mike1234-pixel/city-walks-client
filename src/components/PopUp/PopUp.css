/** @format */

.popup-container-cookies {
  position: relative;
  z-index: 20;
}

.popup-cookies {
  background: rgba(63, 63, 63, 0.842);
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  color: white;
  font-size: 30px;
  padding: 20px;
  display: flex;
  flex-direction: column;
}
