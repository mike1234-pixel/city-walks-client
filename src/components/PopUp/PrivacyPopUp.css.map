{"version":3,"sourceRoot":"","sources":["PrivacyPopUp.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;AAAA;AAAA;EAGE;;;AAGF;AAAA;AAAA;EAGE","file":"PrivacyPopUp.css"}