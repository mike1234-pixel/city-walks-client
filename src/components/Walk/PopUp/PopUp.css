/** @format */

.popup-container {
  position: relative;
}

.popup {
  background: rgba(63, 63, 63, 0.842);
  position: fixed;
  top: 0;
  left: 0;
  z-index: 5;
  width: 100vw;
  height: 100vh;
  display: flex;
}

.popup-img {
  width: 80vw;
  height: 80vh;
  margin: auto;
  border: 2px black solid;
}

@media (min-width: 1000px) {
  .popup-img {
    width: 40vw;
  }
}

/* sample map img */
/* https://coding-video.s3.eu-west-2.amazonaws.com/mapimg.JPG */
