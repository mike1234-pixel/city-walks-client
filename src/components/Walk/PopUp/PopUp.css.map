{"version":3,"sourceRoot":"","sources":["PopUp.scss"],"names":[],"mappings":";AAAA;AAEA;EACE;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;AAEA;EACE;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;IACE;IACA;;;AAIJ;EACE;IACE;IACA;;EAGF;IACE;;;AAIJ;AACA;EACE;IACE;IACA;;;AAIJ;AACA","file":"PopUp.css"}