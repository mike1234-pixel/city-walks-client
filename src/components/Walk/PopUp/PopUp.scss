/** @format */

.popup-container {
  position: relative;
}

.popup {
  background: rgba(63, 63, 63, 0.842);
  position: fixed;
  top: 0;
  left: 0;
  z-index: 5;
  width: 100vw;
  height: 100vh;
  display: flex;
}

/* popup-img height and width in vw to retain aspect ratio */

.popup-img {
  width: 80vw;
  height: 80vw;
  margin: auto;
  border: 2px black solid;
  position: relative;
  z-index: 10;
}

.popup::after {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 30px;
  right: 10;
  content: "\274c \00a0 Hide Map";
  font-size: 30px;
  color: #fff;
  line-height: 100px;
  text-align: center;
  cursor: pointer;
}

@media (min-width: 1000px) {
  .popup-img {
    width: 40vw;
    height: 40vw;
  }
}

@media (max-width: 500px) {
  .popup-img {
    width: 95vw;
    height: 135vw;
  }
  .popup::after {
    left: 1vw;
  }
}

/* ipad pro */
@media (min-width: 1024px) and (max-width: 1025px) and (min-height: 1366px) and (max-height: 1367px) {
  .popup-img {
    width: 65vw;
    height: 65vw;
  }
}

/* sample map img */
/* https://coding-video.s3.eu-west-2.amazonaws.com/mapimg.JPG */
