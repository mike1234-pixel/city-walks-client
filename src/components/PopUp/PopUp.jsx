import { useContext } from "react"
import { LoginContext } from "../../context/LoginContext"
import { MDBBtn } from "mdbreact"
import { Link } from "react-router-dom"
import "./PopUp.css"

const PopUp = () => {

    const { handlePopup } = useContext(LoginContext)

    return (
        <div className="popup-container-cookies">
            <div className="popup-cookies">
                <p>To make City Walks work, we log user data. By using City Walks, you agree to our <Link to="/privacy">Privacy Policy</Link>, including cookie policy.</p>
                <MDBBtn onClick={handlePopup}>Accept</MDBBtn>
            </div>
        </div>
    )
}

export default PopUp