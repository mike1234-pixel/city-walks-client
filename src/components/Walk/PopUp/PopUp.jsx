import './PopUp.css'

const PopUp = (props) => {

    const { mapImg } = props

    return (
        <div className="popup-container">
            <div className="popup">
                {/* orientation will be portrait */}
                <img className="popup-img" src={mapImg}/>      
            </div>
        </div>
    )
}

export default PopUp