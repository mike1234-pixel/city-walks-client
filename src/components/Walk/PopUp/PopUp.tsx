import React from "react";
import { MDBContainer } from "mdbreact";
import "./PopUp.css";

interface Props {
  iframeLink: string;
  iframeTitle: string;
}

const PopUp: React.FC<Props> = (props: Props) => {
  const { iframeLink, iframeTitle } = props;

  return (
    <MDBContainer>
      <div className='popup-container'>
        <div className='popup'>
          {/* orientation will be portrait */}
          <iframe
            src={iframeLink}
            title={iframeTitle}
            className='popup-img'
          ></iframe>
        </div>
      </div>
    </MDBContainer>
  );
};

export default PopUp;
