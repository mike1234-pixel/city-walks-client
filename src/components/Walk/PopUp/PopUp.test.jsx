import renderer from 'react-test-renderer'
import PopUp from './PopUp'

// snapshot test
it('Map Pop Up (in Walk) renders correctly', () => {  

  const mapImg = "mapimg"

  const tree = renderer
    .create(
        <PopUp mapImg={mapImg}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
