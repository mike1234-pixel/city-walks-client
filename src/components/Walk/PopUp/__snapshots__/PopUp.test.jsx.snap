// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Pop Up (in Walk) renders correctly 1`] = `
<div
  className="popup-container"
>
  <div
    className="popup"
  >
    <img
      className="popup-img"
      src="mapimg"
    />
  </div>
</div>
`;
